llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -8832,72 +8901,117 @@ SDValue RISCVTargetLowering::LowerINTRINSIC_VOID(SDValue Op,
         FixedIntrinsic->getMemoryVT(), FixedIntrinsic->getMemOperand());
   }
   case Intrinsic::riscv_sf_vc_x_se_e8mf8:
```

Comment:
Would it be possible to redefine riscv_sf_vc_x_se* and riscv_sf_vc_i_se IR intrinsics store the sew and lmul as immediate operands like vsetvli. Then use those immediate operands during isel to select the instructions? That would remove 42 intrinsics I think.

Reply:
If we redefine `riscv_sf_vc_x_se*` to `riscv_sf_vc_x_se`, we should also change C intrinsics to match those IRs, but we are not able to distinguish between C intrinsics if we don't have suffix. We can make sew and luml as input operands in IR, but we can't add those to C intrinsics, can we?